docs/source/package-spec.rst

```diff
@@ -154,52 +158,43 @@ Sources: ``sources``
 
 The ``sources`` field defines a source tree that **should** comprise the
 full source tree necessary to recompile the contracts contained in this
-release. Sources are declared in a key/value mapping.
+release.
 
+  :Required: No
   :Key: ``sources``
+  :Type: Object (String: `Sources Object`_)
 
-  :Type: Object (String: String)
-
-  :Format: See Below.
-
-Format
-^^^^^^
-
-Keys **must** be relative filesystem paths beginning with a ``./``.
-
-Paths **must** resolve to a path that is within the current working directory.
+----
 
-Values **must** conform to *one of* the following formats.
+Contract Types: ``contractTypes``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-    - Source string.
-    - |ContentAddressableURI|.
+The ``contractTypes`` field hosts the |ContractTypes| which have been
+included in this release.
 
-When the value is a source string the key should be interpreted as a
-file path.
+  - Packages **should** only include contract types that can be found in the source files for this package.
+  - Packages **should not** include contract types from dependencies.
+  - Packages **should not** include abstract contracts in the contract types section of a release.
 
-    - If the resulting document is a directory the key should be
-      interpreted as a directory path.
-    - If the resulting document is a file the key should be
-      interpreted as a file path.
+  :Required: No
+  :Key: ``contractTypes``
+  :Type: Object (String: `Contract Type Object`_)
+  :Format:
+    - Keys **must** be valid |ContractAliases|.
+    - Values **must** conform to the `Contract Type Object`_ definition.
 
 ----
 
-Contract Types: ``contract_types``
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-The ``contract_types`` field holds the |ContractTypes| which have been
-included in this release. |Packages| **should** only include contract types
-that can be found in the source files for this package. Packages
-**should not** include contract types from dependencies. Packages
-**should not** include abstract contracts in the contract types section of a
-release.
+Compilers: ``compilers``
+~~~~~~~~~~~~~~~~~~~~~~~~
 
-  :Key: ``contract_types``
-  :Type: Object (String: `Contract Type Object`_)
-  :Format:
-      Keys **must** be valid |ContractAliases|.
+The ``compilers`` field holds the information about the compilers and their
+settings that have been used to generate the various ``contractTypes`` included
+in this release.
 
-      Values **must** conform to the `Contract Type Object`_ definition.
+  :Required: No
+  :Key: ``compilers``
+  :Type: Array (`the Compiler Information object`_)
```

Comment:
I'm confused by this being an array.  Still reading so maybe it will be self explanitory.